apps/aecore/src/aec_db.erl

```diff
@@ -860,3 +854,24 @@ default_dir() ->
         {ok, Dir} ->
             Dir
     end.
+
+try_activity(Type, Fun) ->
+    %% If no configuration is found, we only retry once.
+    MaxRetries = aeu_env:user_config_or_env([<<"chain">>, <<"db_write_max_retries">>],
+                                            aecore, db_write_max_retries, 1),
+    try_activity(Type, Fun, MaxRetries).
+
+%% @doc Run function in an mnesia activity context. If retries is 0 or less, the
+%% operation will not be retried upon failure. If retries is greater than 0, the
+%% operation will be retried upon failure and the retry counter will be
+%% decremented.
+try_activity(Type, Fun, Retries) when Retries =< 0 ->
+    mnesia:activity(Type, Fun);
+try_activity(Type, Fun, Retries) ->
+    try
+        mnesia:activity(Type, Fun)
+    catch
+        exit:Reason ->
```

Comment:
This is very dangerous. If you catch unqualified stuff like this inside of a mnesia transaction, you disrupt mnesia's internal retry mechanism.

Probably this is all right, but if this happens to be called inside of a (wrapped) transaction, this will get the wrong retry semantics and might corrupt data.

Reply:
Good point. Since this function is called out of `ensure_transaction` which does check that it is not executed inside another transaction, I figured it should be ok. Or should I check here as well to prevent future misuse?

Reply:
You are correct. This is currently safe, at least in the transaction case. I would add a comment about this rather than a check, and skip the `try_activity/2` to make it more obvious that this should not be used from any call point.